File: web/src/renderer/api/apikey.js
import request from './request';

/**
 * API Key：对外 /v1 接口鉴权凭据
 * 单 key 场景：仅「读」与「重新生成」两个动作，返回完整 key 便于复制。
 */

/** 获取唯一 API Key */
export function getAPIKey() {
  return request.get('/apikey');
}

/** 重新生成 API Key，返回新 key 完整明文 */
export function regenerateAPIKey() {
  return request.post('/apikey/regenerate');
}

File: backend/api/apikey.go
package api

import (
	"net/http"
	"time"

	"github.com/gin-gonic/gin"

	"github.com/Lixiuxiu559/portunus/backend/shared"
)

// registerAPIKeyRoutes 注册 API Key 管理路由。
// 单 key 场景：仅提供「读」与「重新生成」两个动作。
func registerAPIKeyRoutes(r *gin.RouterGroup) {
	g := r.Group("/apikey")
	g.GET("", getAPIKey)
	g.POST("/regenerate", regenerateAPIKey)
}

// apiKeyResponse 是 API Key 的对外响应，返回完整 key 便于复制。
type apiKeyResponse struct {
	ID        int64     `json:"id"`
	Key       string    `json:"key"`
	CreatedAt time.Time `json:"created_at"`
}

func toAPIKeyResponse(k *shared.APIKey) apiKeyResponse {
	return apiKeyResponse{ID: k.ID, Key: k.Key, CreatedAt: k.CreatedAt}
}

func getAPIKey(c *gin.Context) {
	k, err := shared.GetAPIKey()
	if err != nil {
		respondError(c, err)
		return
	}
	c.JSON(http.StatusOK, toAPIKeyResponse(k))
}

func regenerateAPIKey(c *gin.Context) {
	k, err := shared.RegenerateAPIKey()
	if err != nil {
		respondError(c, err)
		return
	}
	c.JSON(http.StatusOK, toAPIKeyResponse(k))
}